storage/innobase/handler/ha_innodb.cc

```diff
@@ -857,7 +857,7 @@ static MYSQL_THDVAR_UINT(default_encryption_key_id, PLUGIN_VAR_RQCMDARG,
                          "Default encryption key id used for table encryption.",
                          default_encryption_key_id_validate,
                          default_encryption_key_id_update,
-                         FIL_DEFAULT_ENCRYPTION_KEY, 0, UINT_MAX32, 0);
+                         FIL_DEFAULT_ENCRYPTION_KEY, 0, UINT_MAX32 - 1, 0);
```

Comment:
when we specify a range here, isn't this enough? I haven't seen any other variable which has the special parse code to determine out of range.

Reply:
This range check is for --innodb-default_encryption_key_id option. The yacc changes are for ENCRYPTION_KEY_ID in CREATE/ALTER TABLE/TABLESPACE.

Reply:
I see. Thanks!